### DISCUSSION TITLE
Does this support the tracker?

### DISCUSSION THREAD

**Discussion by Jonnehs** [BottlecapDave/HomeAssistant-OctopusEnergy#833] [Q&A]
Sorry, had a bit of a look around and couldn't see an answer. Do Octopus publish the daily tracker rate on their API, and does this integration pull it if that's the tariff your meter is set to?

Thanks!

**Comment by BottlecapDave** [comment]
They do and this integration does.

**Comment by Jonnehs** [comment]
@BottlecapDave thank you. Great, I made a little automation to sent me a push notification when your octopus_energy_electricity_current_day_rates event fires, I presume this will do so whenever the API rate gets updated?

Is there a template automation that I can't find in the documentation for updating an energy meter object with the rate every day so home assistant can track the costs?

**Reply by BottlecapDave** [reply]
The rates event will fire everytime the rate information is retrieved, which is roughly every 30 minutes. There is a basic template available at https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/events/#electricity-current-day-rates, but can't cover all possibilities of the sensor. This event will obviously include all rates for the current day.

Depending on what you're after, there is already a sensor for providing the [current rate](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/entities/electricity/#current-rate). For a tracker tariff this will update everytime the current rate changes, which you can use in an automation to set the value in something else.

**Reply by Jonnehs** [reply]
Thanks.

I have noticed that the event is a bit spammey as a trigger. I just want a push notification when the rate updates, so I'll need to find a trigger solution that only sends me a push if the rate changes I guess. Thats a bit complex for my code-less copy/paste existence, so if you have any guidance on a good way to handle it I'd appreciate it.

Honestly with the sensor I already knew about the current rate, it just looks a bit wordy on the Energy dashboard because of the meter name/number etc so I just wanted it to update a meter helper object that can have a nice presentable name for the sake of readability and aesthetics. I actually think you can just do that in the helper itself looking into it, using your sensor as the input source?

Thanks for your work, this integration has really come in handy!

**Reply by Jonnehs** [reply]
I actually solved the notification part, was easier than I thought - just trigger from the current rate sensor but ignore attribute changes.

```
alias: Energy - Rate Update Phone Notification
description: ""
trigger:
  - platform: state
    entity_id:
      - sensor.octopus_energy_electricity_{{METER_SERIAL_NUMBER}}_{{MPAN_NUMBER}}_current_rate
    to: null
condition: []
action:
  - service: notify.phones
    metadata: {}
    data:
      title: Octopus Tracker Rate
      message: >-
        Octopus tracker rate for today is {{
        states("sensor.octopus_energy_electricity_{{METER_SERIAL_NUMBER}}_{{MPAN_NUMBER}}_current_rate")
        }}
      data:
        color: yellow
mode: single
```


Just in case anyone finds this on a search and wants a similar solution

**Reply by BottlecapDave** [reply]
As per the FAQ, you can always change the name of the sensor to be something more to your liking if the name is the only reason for this :)
